e2e/testcases/lifecycle_directives_test.go

```diff
@@ -61,23 +60,22 @@ func TestPreventDeletionNamespace(t *testing.T) {
 	nt.Must(rootSyncGitRepo.CommitAndPush("declare Namespace with prevent deletion lifecycle annotation"))
 	nt.Must(nt.WatchForAllSyncs())
 
-	err = nt.Validate("shipping", "", &corev1.Namespace{},
+	rsObj := &v1beta1.RootSync{}
+	nt.Must(nt.KubeClient.Get(rootSyncNN.Name, rootSyncNN.Namespace, rsObj))
+	applySetID := core.GetLabel(rsObj, metadata.ApplySetParentIDLabel)
```

Comment:
Instead of getting applySetID from a GET request, can we use `applyset.IDFromSync`?

Reply:
Yes, but getting it from the RSync is slightly better for a few reasons:
- It validates that the label is set on the RSync
- It validates the ID matches on the RSync and managed objects
- It's more like what customers would do if they try to build automation around ApplySet labels.

The extra API call shouldn't significantly slow down testing.